parsec/core/fs/storage/manifest_storage.py

```diff
@@ -130,40 +130,73 @@ def clear_memory_cache(self):
         Raises:
             FSLocalMissError
         """
+        # Look in cache first
         try:
             return self._cache[entry_id]
         except KeyError:
             pass
 
+        # Look into the database
         async with self._open_cursor() as cursor:
             cursor.execute("SELECT blob FROM vlobs WHERE vlob_id = ?", (entry_id.bytes,))
             manifest_row = cursor.fetchone()
+
+        # Not found
         if not manifest_row:
             raise FSLocalMissError(entry_id)
 
-        manifest = LocalManifest.decrypt_and_load(manifest_row[0], key=self.device.local_symkey)
-        self._cache[entry_id] = manifest
+        # Safely fill the cache
+        if entry_id not in self._cache:
+            self._cache[entry_id] = LocalManifest.decrypt_and_load(
+                manifest_row[0], key=self.device.local_symkey
+            )
 
-        return manifest
+        # Always return the cached value
+        return self._cache[entry_id]
 
     async def set_manifest(
-        self, entry_id: EntryID, manifest: LocalManifest, cache_only: bool = False
+        self,
+        entry_id: EntryID,
+        manifest: LocalManifest,
+        cache_only: bool = False,
+        cleanup: Optional[Set[ChunkID]] = None,
     ) -> None:
         """
         Raises: Nothing !
         """
         assert isinstance(entry_id, EntryID)
 
-        if not cache_only:
-            await self._set_manifest_in_db(entry_id, manifest)
-
-        else:
-            self._cache_ahead_of_persistance_ids.add(entry_id)
+        # Set the cache first
         self._cache[entry_id] = manifest
+        self._cache_ahead_of_localdb.setdefault(entry_id, set())
```

Comment:
pourquoi ne pas faire de `self._cache_ahead_of_localdb` un `defaultdict(set)` du coup ?

Reply:
parce qu'il y a une grande différence entre:
- `entry_id not in _cache_ahead_of_localdb`: la localdb est à jour avec le cache
- `_cache_ahead_of_localdb[entry_id] == set()`: la localdb n'est PAS à jour avec le cache (i.e. il n'y a aucun chunk à supprimer mais le manifest doit quand meme etre ajouté au localdb)

Reply:
mais du coup on peut faire
```python
entry_id in _cache_ahead_of_localdb and not _cache_ahead_of_localdb[entry_id]
```
ou alors
```python
_cache_ahead_of_localdb.get(entry_id, None) == set()
```

après je pense que le plus important c'est de mettre un petit commentaire à l'endroit de la définition de `_cache_ahead_of_localdb` pour expliquer qu'il a deux rôles (genre `Used to keep track of entries to sync AND their related chunk that need to be cleaned up on sync`)

Reply:
> après je pense que le plus important c'est de mettre un petit commentaire à l'endroit de la définition

Et voila : 1f7073d :)